src/main/java/apoc/atomic/Atomic.java

```diff
@@ -0,0 +1,250 @@
+        checkIsPropertyContainer(container);
+        PropertyContainer propertyContainer;
+        final AtomicReference oldValue = new AtomicReference();
+        final AtomicReference newValue = new AtomicReference();
+        propertyContainer = (PropertyContainer) container;
+        final ExecutionContext executionContext = new ExecutionContext(db, propertyContainer, property);
+
+        retry(executionContext, (context) -> {
+            Object[] arrayBackedList = new ArrayBackedList(propertyContainer.getProperty(property)).toArray();
+            oldValue.set(arrayBackedList);
+            if(position > arrayBackedList.length || position < 0) {
+                throw new RuntimeException("Attention your position out of range or higher than array length, that is " + arrayBackedList.length);
+            }
+            Object[] newArray = ArrayUtils.addAll(Arrays.copyOfRange(arrayBackedList, 0, position.intValue()), Arrays.copyOfRange(arrayBackedList, position.intValue() +1, arrayBackedList.length));
+            Class clazz;
+            try {
+                clazz = Class.forName(arrayBackedList[0].getClass().getName());
+            } catch (ClassNotFoundException e) {
+                throw new RuntimeException(e);
+            }
+            /*it's not possible to return directly the newArray, we have to create a new array with the specific class*/
+            newValue.set(Array.newInstance(clazz, newArray.length));
+            System.arraycopy(newArray, 0, newValue.get(), 0, newArray.length);
+            propertyContainer.setProperty(property, newValue.get());
+
+            return context.propertyContainer.getProperty(property);
+        }, times);
+
+        return Stream.of(new AtomicResults(propertyContainer, property, oldValue.get(), newValue.get()));
+    }
+
+    /**
+     * update the property's value
+     */
+    @Procedure(mode = Mode.WRITE)
+    @Description("apoc.atomic.update(node/relatonship,propertyName,updateOperation) update a property's value with a cypher operation (ex. \"n.prop1+n.prop2\")")
+    public Stream<AtomicResults> update(@Name("container") Object container, @Name("propertyName") String property, @Name("operation") String operation, @Name(value = "times", defaultValue = "5") Long times) throws InterruptedException {
+        checkIsPropertyContainer(container);
+        PropertyContainer propertyContainer = (PropertyContainer) container;
+        final AtomicReference oldValue = new AtomicReference();
+        final ExecutionContext executionContext = new ExecutionContext(db, propertyContainer, property);
+
+        retry(executionContext, (context) -> {
+            oldValue.set(propertyContainer.getProperty(property));
+            String statement = "UNWIND {container} as n with n set n." + property + "=" + operation + ";";
```

Comment:
UNWIND is not needed as the object is not a list
a `WITH {container} as n` should be good enough, but property should be backticked.
Also mention in the docs that the container is `n`.